src/parser/rogue/outlaw/modules/talents/QuickDraw.js

```diff
@@ -18,27 +18,40 @@ class QuickDraw extends Analyzer {
     if (!this.active) {
       return;
     }
-    this.addEventListener(Events.energize.by(SELECTED_PLAYER).spell(SPELLS.PISTOL_SHOT), this._onPistolShot);
-    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.PISTOL_SHOT), this._onPistolShot);
+    this.addEventListener(Events.energize.by(SELECTED_PLAYER).spell(SPELLS.PISTOL_SHOT), this._onPistolShotEnergize);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.PISTOL_SHOT), this._onPistolShotDamage);
   }
 
   damageGained = 0;
-  comboPointsGained = 0;
+  procs = 0;
+  comboPointsWasted = 0;
 
-  _onPistolShot(event) {
+  _onPistolShotEnergize(event) {
     if (!this.selectedCombatant.hasBuff(SPELLS.OPPORTUNITY.id)) {
       return;
     }
-    this.comboPointsGained +=
-      (event.resourceChange > 0 ? event.resourceChange - BASE_CP_GENERATED : 0) || 0;
+    const cpGeneratedByBroadside = this.selectedCombatant.hasBuff(SPELLS.BROADSIDE.id) ? 1 : 0;
+    const cpForEvent = (event.resourceChange || 0) + (event.waste || 0) - cpGeneratedByBroadside - BASE_CP_GENERATED;
+    this.procs += cpForEvent > 0 ? 1 : 0;
+    if(cpForEvent>0 && event.waste>0){
+        this.comboPointsWasted+=1;
```

Comment:
`comboPointsWasted` seems to be calculated but not used. I suggest removing it to keep it simple